cdap-api/src/main/java/co/cask/cdap/api/annotation/Capability.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright © 2018 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package co.cask.cdap.api.annotation;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * <p>Annotates capabilities of a plugin.</p>
+ * <p>Pre-defined system capabilities are defined in {@link Capabilities}.</p>
+ * <p>Plugins can also be annotated with custom value to specify capability with business rules.
+ * The capability value is case insensitive.</p>
+ * <p>If a plugin does not specify any capability or the {@link #value()} array is empty then the default capability
+ * will be used.</p>
+ * <p>According to default capability the plugin will be considered as capable of all system defined
+ * {@link Capabilities} but not with any user defined capability options.
+ * Custom capabilities needs to be specifically defined.</p>
+ * <p>Usage Examples:</p>
```

Comment:
It would be good to have an example where no Capability is specified.

Reply:
Done added more usage examples.